torchvision/prototype/datapoints/_dataset_wrapper.py

```diff
@@ -201,9 +204,12 @@ def segmentation_to_mask(segmentation, *, spatial_size):
         )
         return torch.from_numpy(mask.decode(segmentation))
 
-    def wrapper(sample):
+    def wrapper(idx, sample):
         image, target = sample
 
+        if not target:
+            return image, dict(image_id=dataset.ids[idx])
```

Comment:
Do we need to return more than the image ID here?

Reply:
I don't think we *need* to for our purposes, but I'd suggest something more like what I did in https://github.com/pytorch/vision/pull/7220/files#diff-6ecb3203d81a2fa8526df6470e135033f8907ed4626faa0f3a44890b7b22f526R216

(see the protection against missing keys, and the fact taht we retrieve the image_id from the dataset.ids regardless of a missing target)

Reply:
> see the protection against missing keys

That looks less safe to me. If either "bbox" or "segmentation" is missing from the target, something has gone horribly wrong. In such a case IMO it is totally fine to fail loudly. What use case are you imagining where this could happen or the behavior would be useful?

> and the fact taht we retrieve the image_id from the dataset.ids regardless of a missing target

IMO both variants are equally safe. Somewhat related to above: what use case do you have in mind where the annotations don't store the image ID?

Reply:
I just don't know COCO enough to be sure this doesn't lead to an error. What if there's a weirdly built validation sample that has a non-empty `annotation` dict, but doesn't have an `image_id` key? Or one that has bboxes but no masks?
If we can validate those assumption on the whole dataset, then that's fine. 

Although, I'd still recommend having one single logic for accessing `image_id` instead of 2

Reply:
I've verified on the complete validation split for 2017 and if an annotation exists, it is always complete.